subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseModel.java

```diff
@@ -67,6 +71,8 @@
 
     private EclipseWtp wtp;
 
+    private List<String> synchronizationTasks = new ArrayList<>();
```

Comment:
Use a `DefaultTaskDependency` as your internal model here, so you can resolve any notation that would be accepted by `Task.dependsOn`

Reply:
Thanks for the advice, Fixed. (I've learned a lot about the codebase via this PR).